build fails with txt2man installed

I was trying to build `mlpack` with the following:
```
cmake -DBUILD_CLI_EXECUTABLES=OFF -DBUILD_TESTS=OFF -DBUILD_PYTHON_BINDINGS=OFF ..
make
```
Although `cmake` succeeded, but `make` was complaining with
```
Scanning dependencies of target mlpack_gitversion
[  1%] Updating gitversion.hpp (if necessary)
-- Found Git: /usr/bin/git (found version "2.7.4") 
-- gitversion.hpp is already up to date.
[  1%] Built target mlpack_gitversion
Scanning dependencies of target mlpack_arma_config
[  2%] Updating arma_config.hpp (if necessary)
-- Regenerating arma_config.hpp.
[  2%] Built target mlpack_arma_config
Scanning dependencies of target man
make[2]: *** No rule to make target '../mlpack_sparse_coding', needed by 'CMakeFiles/man'.  Stop.
CMakeFiles/Makefile2:173: recipe for target 'CMakeFiles/man.dir/all' failed
make[1]: *** [CMakeFiles/man.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```

I realized that the `CMake code` that is making trouble is inside the `else part` of
```
if (NOT TXT2MAN)
    message(WARNING "txt2man not found; man pages will not be generated.")
  else ()
    . . .
endif ()
```

so, I removed the `txt2man` from my system and tried to build again and it **succeeded** .

**System Information:** Ubuntu 16.04, CMake 3.11, gcc/g++ 7.3

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -479,7 +479,7 @@ endif ()
 
 # Make a target to generate the man page documentation, but only if we are on a
 # UNIX-like system.
-if (UNIX)
+if (BUILD_CLI_EXECUTABLES AND UNIX)
   find_program(TXT2MAN txt2man)
 
   # It's not a requirement that we make man pages.

